lib/src/Base/Graph/DrawableImplementation.cxx

```diff
@@ -816,37 +816,20 @@ String DrawableImplementation::ConvertFromName(const String & name)
 /* Convert an hexadecimal code into an RGB triplet */
 Indices DrawableImplementation::ConvertToRGB(const String & key)
 {
-  Indices rgb(3, 0);
-  UnsignedInteger code = 0;
-  if (ScanColorCode(key, code))
-  {
-    code = code >> 8;
-    rgb[2] = code % 256;
-    code = code >> 8;
-    rgb[1] = code % 256;
-    code = code >> 8;
-    rgb[0] = code;
-  }
-  else LOGWARN(OSS() << "Code " << key << " is an invalid color code. Default to black.");
-  return rgb;
+  Indices rgba;
+  if (!ScanColorCode(key, rgba))
+    LOGWARN(OSS() << "Code " << key << " is an invalid color code. Default to black.");
+  // Drop the a component
+  rgba.resize(3);
+  return rgba;
 }
 
 /* Convert an hexadecimal code into an RGBA quadruplet */
 Indices DrawableImplementation::ConvertToRGBA(const String & key)
 {
-  Indices rgba(4, 0);
-  UnsignedInteger code = 0;
-  if (ScanColorCode(key, code))
-  {
-    rgba[3] = code % 256;
-    code = code >> 8;
-    rgba[2] = code % 256;
-    code = code >> 8;
-    rgba[1] = code % 256;
-    code = code >> 8;
-    rgba[0] = code;
-  }
-  else LOGWARN(OSS() << "Code " << key << " is an invalid color code. Default to black.");
+  Indices rgba;
+  if (!ScanColorCode(key, rgba))
+    LOGWARN(OSS() << "Code " << key << " is an invalid color code. Default to black.");
   return rgba;
```

Comment:
Maybe this could be reused to factorize some code with the PolygonArray section where color conversion seems partially rewritten:
```
elif drawableKind == 'PolygonArray':

                polygonsNumber = drawable.getPalette().getSize()
                verticesNumber = drawable.getData().getSize() // polygonsNumber
                colors = drawable.getPalette()
                def convRGB(c):
                    return (int(c[1:3], 16) / 255.0, int(c[3:5], 16) / 255.0, int(c[5:7], 16) / 255.0, 1.0)
                def convRGBA(c):
                    return (int(c[1:3], 16) / 255.0, int(c[3:5], 16) / 255.0, int(c[5:7], 16) / 255.0, int(c[7:9], 16) / 255.0)
                if colors[0][0] == '#' and len(colors[0]) == 7:
                    colorsRGBA = [convRGB(color) for color in colors]
                elif colors[0][0] == '#' and len(colors[0]) == 9:
                    colorsRGBA = [convRGBA(color) for color in colors]
                else:
                    colorsRGBA = []
                    for i in range(polygonsNumber):
                        hex_code = ot.Drawable.ConvertFromName(colors[i])
                        rgba = ot.Drawable.ConvertToRGBA(hex_code)
                        colorsRGBA.append(
                            (rgba[0] / 255.0, rgba[1] / 255.0, rgba[2] / 255.0, rgba[3] / 255.0))
                if 'facecolors' not in polygoncollection_kwargs_default:
                    polygoncollection_kwargs['facecolors'] = colorsRGBA

                if 'edgecolors' not in polygoncollection_kwargs_default:
                    polygoncollection_kwargs['edgecolors'] = colorsRGBA
                self._ax[0].add_collection(
                    matplotlib.collections.PolyCollection(np.array(data).reshape((polygonsNumber, verticesNumber, 2)), **polygoncollection_kwargs))
```

Reply:
The problem is that no C++ method is provided to convert the whole palette with a single call.

Reply:
I added Drawable::getPaletteAsNormalizedRGBA() to solve this problem. Any other issue here?